confirm-deposit.md

```diff
@@ -0,0 +1,109 @@
+# Confirm a Deposit to the Exchange
+
+Deposits are made in several stages. 
+One of the stages is to go through the [Exchange server](https://docs.flare.network/exchange/architecture/#architecture-of-an-exchange), which monitors for deposits and updates the Balances database. 
+Run the code below to have the Exchange server listen for deposit transactions and confirm the deposits.
+
+On a blockchain, multiple [validators](h[ttps://docs.flare.network/tech/glossary/](https://docs.flare.network/tech/validators/)) examine newly created transactions and come to an agreement about whether they are valid. 
+When they are valid, they are added to a new block on the chain and identified by a one-way cryptographic algorithm, called a _hash_. 
+The one-way hash prevents transactions from being reverted. 
+[Blocks](https://docs.flare.network/tech/glossary/) are added to the [blockchain](https://docs.flare.network/tech/glossary/) in chronological order and it may take a few blocks for the hash to be completed.
+To prevent problems, we confirm deposits several blocks back (five blocks in our code), when the blocks are most likely to be irreversible.
```

Comment:
And the final piece of the puzzle. Waiting for 100% agreement of all validators would be too slow (imagine you pay with your credit card on a shop and you are waiting for the payment to be confirmed. You want this to be quick!).
Therefore, transactions are usually accepted **before** there's full consensus. This means that people on different parts of the room might reach different conclusions on whether your transaction was accepted or not. These are called different _branches_ of the blockchain.
Later, since these people are constantly moving around the room and talking to each other, some might realize that different conclusions were reached for a past block and they'll fix it. Fixing it means that one of the branches is discarded and its followers jump to the other one. And in turn, this means that some transactions that had already been accepted in the discarded branch are now not accepted anymore. THIS is where reverted transactions come from.
And the way to avoid this is to wait for a few blocks, so that all validators have had a chance to look at your transaction and you are sure it will not be reverted.

Reply:
Awesome, I'm learning some more detail about how validators work.